pkg/registry/core/pod/storage/eviction.go

```diff
@@ -226,10 +226,26 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		pdb := &pdbs[0]
 		pdbName = pdb.Name
 
-		// If the pod is not ready, it doesn't count towards healthy and we should not decrement
-		if !podutil.IsPodReady(pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
-			updateDeletionOptions = true
-			return nil
+		// IsPodReady is the current implementation of IsHealthy
+		// If the pod is healthy, it should be guarded by the PDB.
+		// If the pod is not healthy, it doesn't count towards currentHealthy and desiredHealthy and we should not decrement disruptionsAllowed.
```

Comment:
move the `it doesn't count towards currentHealthy and desiredHealthy and we should not decrement disruptionsAllowed` comment to the two places inside this block where we return nil... when we don't return and continue on, we *do* decrement `disruptionsAllowed`

Reply:
+1, changed